staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -37,12 +37,34 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
-// StartUntil starts f in a new goroutine and calls done once f has finished.
-func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
+// Group is an interface to decouple code from sync.WaitGroup.
+type Group interface {
+	Add(delta int)
+	Done()
+}
+
+// StartWithChannelWithinGroup adds 1 to the group, starts f in a new goroutine and calls g.Done once f has finished.
+// stopCh is passed to f as an argument.
+func StartWithChannelWithinGroup(stopCh <-chan struct{}, g Group, f func(stopCh <-chan struct{})) {
```

Comment:
Fair enough, though I think that you're reiterating [my initial complaint](https://github.com/kubernetes/kubernetes/pull/46094#discussion_r119633960). You could meet halfway and mention a stop channel in the documentation as the motivating case, but leave the parameter names noncommittal.

It's a small thing, and not worth holding up the merge. I appreciate that you already took this much of my feedback seriously.